contrib/objectstore/subscriber/flowclassifier.go

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package subscriber
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/skydive-project/skydive/flow"
+	"github.com/skydive-project/skydive/logging"
+)
+
+// FlowClassifier classifies flows to different tags (strings)
+type FlowClassifier interface {
+	// GetFlowTag returns the tag of the given flow
+	GetFlowTag(fl *flow.Flow) Tag
+}
+
+// Tag represents the flow classification
+type Tag string
+
+const (
+	tagOther    Tag = "other"
+	tagEgress   Tag = "egress"
+	tagIngress  Tag = "ingress"
+	tagInternal Tag = "internal"
+)
+
+// FlowDirectionClassifier classifies flows by their direction (ingress, egress, etc)
+type FlowDirectionClassifier struct {
+	clusterNetMasks []*net.IPNet
+}
+
+// GetFlowTag tag flows based on src and dst IP ranges
+func (fc *FlowDirectionClassifier) GetFlowTag(fl *flow.Flow) Tag {
+	if fl == nil || fl.Network == nil {
+		return tagOther
+	}
+	isSrcInCluster, err := fc.isClusterIP(fl.Network.A)
+	if err != nil {
+		return tagOther
+	}
+	isDstInCluster, err := fc.isClusterIP(fl.Network.B)
+	if err != nil {
+		return tagOther
+	}
+
+	if isSrcInCluster {
+		if isDstInCluster {
+			return tagInternal
+		}
+		return tagEgress
+	}
+
+	if isDstInCluster {
+		return tagIngress
+	}
+	return tagOther
+}
+
+// isClusterIP check if IP is in defined subnet
+func (fc *FlowDirectionClassifier) isClusterIP(ip string) (bool, error) {
```

Comment:
can be simplified to prototype  `isClusterIP(ip string) bool` returning true iff ip is both valid and within at-least one cluster-netmask.

Reply:
This will change the semantics of `GetFlowTag`, so I don't think I can follow your suggestion.

Reply:
are you ever making use of the returned `error`? if not then maybe its' redundant.